### DISCUSSION TITLE
2024-06-07 - 3 steps to a PBI - Chewing the Fat 📈

### DISCUSSION THREAD

**Discussion by bradystroud** [SSWConsulting/SSW.Rules.Content#8677] [Chewing the Fat]
https://www.ssw.com.au/rules/3-steps-to-a-pbi/

What the **most** important step to you and why?
What the **least** important step to you and why?
**What ones do you commonly miss 🫣?**

**Comment by PothieuG** [comment]
- The first step, breaking down the PBI into clear and actionable tasks, is very important as it can significantly enhance work efficiency.
- I can't choose the least important step 😅
- I miss to close the PBI...

**Comment by tkapa** [comment]
Most important - Time to Code - Getting a checked by / feedback
Least Important - (Visibly) Breaking down PBIs into tasks. This should probably be done when the PBI is refined.

Most missed - Closing with Context. I find that the Closed PR being tracked to the PBI is generally good enough to provide the context.

**Reply by Freego1783** [reply]
What he said!

**Reply by GordonBeeming** [reply]
Time to Code feels least important to me, because any complex thinking and planning is done before then

**Comment by ricksu978** [comment]
### Most Important
Make sure your code is deployed 🚀

If the code is not in Production then basically you have done nothing.

### Least Important
Done Video

To me, this only applies to complicated PBIs that are hard to explain in writing.
For simple PBIs, I wouldn't do a Done Video.

### Most Missed
Closing with Context.

In GitHub PR, I use [keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to auto-close the issue.
But I do agree the issue should be closed with context.

**Comment by tiagov8** [comment]
I value the 3 steps equally in importance.

I think the rule refers to ideal conditions for a perfect PBI. In the real world many simpler PBIs won't require all the steps (e.g. the Done Video only necessary in particular scenarios).

The greatest outcome is that PBI must communicate well. So it should be easy for any developer or team member to understand it's goal and status when they look at it.

**Comment by brookjeynes-ssw** [comment]
> What the most important step to you and why?

Step 1. If it's not defined correctly, hours of dev time could be wasted.

> What the least important step to you and why?

N/A, they're all important within the process.

> What ones do you commonly miss 🫣?

Closing issues with context

**Comment by danielmackay** [comment]
> What the most important step to you and why?

Showing Product Owner + Actioning Feedback.  The most important part of development is working software.  Showing the product owner is a great way to check you're on the right path, and make your work event better!

However, for this to work well the product owner needs to be available.

> What the least important step to you and why?

Choosing a PBI.  All PBIs in a Sprint need to be done.  Just grab one!

> What ones do you commonly miss 🫣?

Done video 🙈

**Comment by drwharris** [comment]
So many different opinions on what should be what. In the end, so long as the user requirements are documented and the work to be done has been thought through and broken down into something that is relevant to the audience then it doesnt really matter.

**Comment by joshbermanssw** [comment]
"Ready" is the most important to me - as most wasted time or problems arise when developers arent correctly across what their tasks are 

"Done" - would be the least important of the three to me but i still think this is a crucial step

Usually closing with enough context (i.e screenshots, diagrams, videos)

**Reply by leahy268** [reply]
I'm with Josh here.  I don't think any 1 is much more important than another.  But for me the most important is the Ready.  Only because I find that if the task is not clear then a lot of time can be wasted working it out.

**Comment by bettybondoc** [comment]
> What the most important step to you and why?

I think they’re all important but perhaps step 2 because that is the chunk of the work (coding, testing, feedback)

> What the least important step to you and why?

I don’t think there is one. Step 2 is just slightly more important than the other 2 but we need all 3 steps I believe.

> What ones do you commonly miss 🫣?

Probably step 1, esp the breaking down of the tasks. Especially when I get a very urgent request (due yesterday) and have to jump on the task quickly, I tend to improvise and break down the task as I design.

**Comment by Calinator444** [comment]
> What the most important step to you and why?

Others in this thread have already touched on this point but again, I'd say Done is the most important step. It's the most difficult step to achieve. Really this step alone is 3 steps because you need to:

1. Get your code working
2. Make adjustments to your code if it's not up to a sufficient standard
3. Deploy your changes (assuming this is par of your definition of done)


> What the least important step to you and why?

All of these are important steps. I would say that Done videos are the least important relatively speaking because they aren't always required to complete a PBI. But they're still a great tool for demonstrating the value you've provided to your product owner.

> What ones do you commonly miss 🫣?

Closing the PBI. It's easy to miss when deployment is part of your definition of done because you can't use keywords in GitHub to automatically close you're PBIs.

**Comment by matt-goldman-ssw** [comment]
> What the most important step to you and why?

PBI ready. If we don't have a solid definition of ready, we can waste time on tasks that are not ready to work on. This can be costly for the client and lead to friction (or worse).


> What the least important step to you and why?

Step - done. When it's done, it's done. Of course it's important to have a DoD, but this should be the part that you spend the least amount of time on.
In terms of the rule? The coding step. It's the most important step of the work, but we have a lot of rules covering coding; it can be light on in this one.


> What ones do you commonly miss 🫣?

Done. As you can tell from my above response I have a terrible attitude toward this part that I need to work on.

**Comment by uly1** [comment]
being aware that the DoD includes testig and deployment is the biggest thing

**Comment by Tanya-Leahy** [comment]
It's important to document as much as possible in a PBI, for reference now and to keep a history of what was done when reviewed at a later stage. This saves so much time in the long run.

**Comment by SamBWagner** [comment]
This Rule very much over-reaches the bounds of reasonability. 

PBI's are just meaningful work that can be completed in a timebox. The minutia surrounding their exact shape is not of great value.

So long as that PBI clearly communicates to the developer what the Product Owner wants and is filled out enough for the developer to build from. 

Some teams, with some products, need super verbose PBI's with tonnes of AC and state management. Others just need a title with whether it's being worked on or not. 

Classic consultant answer: "It depends"

**Comment by jakebayliss** [comment]
What the most important step to you and why?
Making sure the PBI is ready with enough info and AC
What the least important step to you and why?
Replying Done - Generally we will show the done PBI's in the review and move them to done so a little pointless to also comment
What ones do you commonly miss 🫣?
Closing off

**Comment by TobyChurches** [comment]
> What the most important step to you and why?

Breaking the task down - This is important for the dev but more so for the team. I think its essential that PBI progress is transparent. This allows for changes in the Dev availability, meaning that anyone could pick up the task.

> What the least important step to you and why?

I think that all the steps are important. Perhaps done videos - This is important for large presentable features but it is definitely not always applicable.

> What ones do you commonly miss 🫣?

Done videos

**Comment by zacharykeeping** [comment]
> What the most important step to you and why?

Probably step 1 - getting the PBI broken down and planned out correctly means being able to work on items efficiently with less wasted time.

> What the least important step to you and why?

Step 3 - getting the work done is obviously important but this step naturally happens and requires less work than the other two.

> **What ones do you commonly miss 🫣?**

Probably closing off issues with context - I might miss this until the next Sprint Review comes around.

**Comment by GabbyWard** [comment]
- What the most important step to you and why?
Time to Code - It's where the most actions are stored and where the most action of the process is. 
- What the least important step to you and why?
PBI is Ready - Seems like common sense to address Product Backlog and assigning yourself items + breaking down a PBI into tasks might not be necessary.
- What ones do you commonly miss 🫣?
Unsure, haven't done one...yet

**Comment by awaltos** [comment]
> Most important (sub)step? (all 3 *steps* are as important as each other)

Checking DoR and DoD - if the project is well documented then these steps will reduce assumptions and save time. Although there's no mention of Acceptance Criteria which I assume is part of DoR and meant to be followed in the "Code Code Code!" task

> Least important?

Done Video but not because it's the least important, because it can add a lot of overhead. I would replace this with adding screenshots or a Done Gif, reserving a full video for larger PBIs or epics.

> Commonly miss

Sometimes I miss "Reply Done" - mainly because merging a linked PR closes an issue and I might forget to go back and do this task and then scramble to get them all updated at the start of a review 😅

**Comment by steven0x51** [comment]
> What the most important step to you and why?
Getting the PBI Ready is the most important step. As without a well defined PBI, the following steps are meaningless.  
>What the least important step to you and why?
Don't think there's least important
>What ones do you commonly miss?
It's possile to miss "replying done"

**Comment by stephanfakossw** [comment]
Getting your PBI's correct is an essential task 😊

**What the most important step to you and why?**
It's important to have a good definition and acceptance criteria

**What the least important step to you and why?**
Breaking up the PBI into smaller tasks.

**Comment by jeoffreyfischer** [comment]
**Most important:** Assign PBI to yourself. If not, the risk is having another Dev working on the same PBI.
**Less important:** Marking it Done at the end. This will be fixed during the Daily Scrum.
I usually **forget** to ask preliminary feedback to the PO (I wait until the Sprint Review for feedback)

**Comment by duanxinhuan** [comment]
Most important:
I hate to say because I'm lazy to do it - Done video.
Simply because stake holder likes it, it's a good vision memory, it's good we have something to interact with, it's good if I do some fun stuffs in video that will make the team be together better.

The point of done video is not just a demo, but also we find something together to do as a client and a consultancy.

Can we have a video for done video?


I'm still sometimes too lazy to do it, but I just can't deny its value.

**Comment by amankumarrr** [comment]
PBI is Ready is the most important step - It must state the Definition of Done and acceptance criteria 
Less important: Done video for every issue 
I usually forget to add the context for the Issue as in GitHub when your PR gets merged, it closes the issue as well.

**Comment by cvionnet** [comment]
- What the most important step to you and why?
"Step 1. PBI is Ready": Without proper preparation, any plan is likely to go wrong
_"By failing to prepare, you are preparing to fail." — Benjamin Franklin_

- What the least important step to you and why?
All the steps are important as they are interconnected

- What ones do you commonly miss 🫣?
Reply 'Done' in the email thread

**Comment by jaydenalchin** [comment]
> Most important 

Step 1 (Ready) - It's key to have clearly defined PBIs that meet the definition of ready 

> Least important 

Step 3 (Done) - This is obviously important, but when the PBI is Done changing the PBI status communicates that, and the PBI itself communicates what was done. I dislike the way this suggests violating DRY and adding unnecessary overhead (Move PBI to Done, comment Done in the PBI with extra details, reply Done to an email with extra details, maybe even a Done video).

> Commonly missed 

Not every In Progress PBI involves Code :p

**Comment by will-truscott-ssw** [comment]
**Most important**
`The breaking down of a PBI into clear actionable tasks. Without this step, miscommunication would not be captured early and development hours may be spent on something that the product manager or client didn't intend.`

**Least important** 
`The Done Video, while invaluable in many contexts, I can see some situations in which during simple and well understood PBIs this could consume valuable dev time.`

**Commonly missed** 
`This process is still new to me, so fortunately I haven't missed anything yet!`

**Comment by YalalCakirSSW** [comment]
> What the most important step to you and why?

PBI is ready - PBI should have all the necessary information and clear acceptance criteria so that any developer can understand the issue or feature.

> What the least important step to you and why?

Done - It is still an important step, especially checking if the PBI is deployed. But I think we should be giving context throughout the whole process and in PR, not just in the end. 

> What ones do you commonly miss 🫣?

Done videos

**Comment by christoment** [comment]
> What the most important step to you and why?

The pre-step: "Before it's ready".
The reason is because we need the PBI to be ready with the proper acceptance criteria, and signed off by the product owner.
In this way, the PBI is "done" when the acceptance criteria is met - any changes to the acceptance criteria warrants a new PBI. This protects the developer from scope creep 😄 

> What the least important step to you and why?

"3. Done". Not least important, but least effort in the grand scheme of things, because the large part of the work is in the first two parts.

😄

**Comment by chrisschultzssw** [comment]
> What the most important step to you and why?

Getting an over the shoulder check/feedback

> What the least important step to you and why?

Done videos - as long as the change is well documented, and you can live-demo it

> What ones do you commonly miss 🫣?

Done videos - they are often not needed for SysAdmin tasks.

**Comment by MarcusIrmscherSSW** [comment]
This rule is very important because it streamlines the process of completing PBIs.

**Comment by tombui99** [comment]
Good set of rules when it comes to completing PBIs.

**Comment by AttackOnMorty** [comment]
> What the most important step to you and why?

PBI is Ready - PBIs should be well-refined and have a clear scope.

**Comment by bradystroud** [comment]
Hot take 🔥: Linking a PR to a PBI is not enough context to close it.

The "Done" comment should show off your work in **production**.
As devs we should be proud of our work and excited to share it with the world. Leaving a nice comment with a little video or a screenshot is an opportunity to do this, yet it's so common this step is completely skipped.

**Comment by alexleessw** [comment]
Most important!
Step 1 - Understanding what needs to be done & breaking down the PBI into clear and actionable tasks

**Comment by Ben0189** [comment]
What the most important step to you and why?
The most important step is showing the PO what has been done and getting early feedback. This helps us align closely with their expectations as consultants.

What the least important step to you and why?
Close the PBI is the least important part , If I had to skip a step, I would choose this one over the others.

**Comment by AntPolkanov** [comment]
> What the most important step to you and why?

Time to code as it brings the actual value.

> What the least important step to you and why?

Couldn't decide.

> What ones do you commonly miss 🫣?

Closing with context. If there is a PBI linked, I don't think the context is required.
If I close a PBI without a PR, e.g. when it's a duplicate or not relevant anymore, then a comment is crucial.

**Reply by bradystroud** [reply]
> If there is a PBI linked, I don't think the context is required.

Linking a PR is good, but it doesn't show the PBI done in production (where it matters)

**Comment by Marxoz** [comment]
Most important -PBI is ready! a PBI should not be pulled into a sprint if it is not ready as per the definition of ready. It will affect the sprint and sprint goals if it is not ready.

Least important - Time to Code. If the PBI is defined correctly, broken up, and in a ready state, coding should be straight forward

None

**Comment by miichaelsmedley** [comment]
Great Job on the Rule.

 It’s important to remember that getting stuff done effectively is our bread and butter. While it’s tempting to speed through tasks, let’s commit to following our updated processes to ensure quality and consistency. I became a scrum master within SSW and love the discipline of task management.

**Comment by bahjatalaadelssw** [comment]
1. Most important - Double check definition of ready. Just because it's approved doesn't mean all the assets are sourced and ready to use for the task. Make sure your resources are also ready.
2. least important - the done video, because list is repetitive, done videos are time consuming so why would you wait to get feedback from the PO only to re-record. I recommend have an over the shoulder with a peer and then the PO before a done video.
3. Miss adding all the tasks

**Comment by camillars** [comment]
> What the most important step to you and why?

PBI is ready - it adds clarity to what steps you will need to do to fully complete the task.

> What the least important step to you and why?

Done - Not least important, but just because it's the easiest. 

> What ones do you commonly miss 🫣?

Done 😅 in Marketing with our intense workload we sometimes take a while to get around to the done when there are many steps to complete e.g. for an event

**Comment by tomek-i** [comment]
>What the most important step to you and why?

Getting feedback early on is important especially if the work is not clearly defined and direction can be adjusted if necessary. When feedback comes in we need to make sure to avoid scope creep under the PBI.

>What the least important step to you and why?

Couldn't really decide which one. Done Videos perhaps because - In most projects most of the work is a lot of internal plumping work and not really something to showcase. It makes sense when a new feature is added to explain the different steps that then can serve as a tutorial. I don't think we need Done video for everything all the time - over the shoulder with the PO when its deployed and or sprint retro should be fine.

>What ones do you commonly miss 🫣?

Adding a closing "Done" comment as I felt it is redundant. I did add closing comments when additional context was required or screenshots.

Like responding to the CTF email with "Done" after filling out the Form - that something i regularly forget.

**Comment by Hona** [comment]
Where is QA in the mix here?
Where is the PO in the sign off for Done? e.g. Michael Jackson rule.

**Comment by JackDevAU** [comment]
The 3 Steps to a PBI poster looks good! 

> 1. PBI is Ready

This section seems a little light for how important a step this one should be. This is the first line of actioning correct work. If this is done incorrectly, you can fall down the rabbit hole of modifying and re-working the PBI to meet the "actual" requirements.

**Comment by MattParkerDev** [comment]
They are all the most important steps.

**Comment by Dhruv-0987** [comment]
Most important - getting a reviewed as much as I can and make changes ( sometime the changes suggested are so good blows my mind that I didn't think of that )
Least important - breaking it into tasks, don't do that more often I have what I need to do in my head and just start coding
Missed - sometimes done video is not required or there no UI to showcase

**Comment by Aibono1225** [comment]
> What the most important step to you and why?

PBI is Ready - Very important as it makes sure that devs won't waste their time on something that isn't viable.

> What the least important step to you and why?

Done Video - I think this should only apply to complicated PBIs that are hard to explain/understand

> What ones do you commonly miss 🫣?

Break down that PBI into clear and actionable tasks

**Comment by jerryluossw** [comment]
> What the most important step to you and why?

Step 1: PBI is Ready.
Because at this step, the analysis of the problem and the design of the solution need to be completed, we also need to determine definition of Done.

> What the least important step to you and why?

Step 3: Done.
It will take me a lot of time if I need to record the done video.

**Comment by ozair-ashfaque** [comment]
> What the most important step to you and why?

PBI is ready - It ensures the direction and breakdown of the PBI  are correct before starting work on it and this step will impact all subsequent steps.

> What the least important step to you and why?

Done - I wouldn't say that this is least important but it requires less effort if all other steps are done efficiently.

> What ones do you commonly miss 🫣?

Videos - I prefer gif and screenshots because they are quicker for smaller features.

**Comment by KristenHu** [comment]
> What the most important step to you and why?

PBI's ready - It may waste lots of time for devs if the PBIs are not ready yet.
> What the least important step to you and why?

Videos - It's not every PBI needs video I think.
> What ones do you commonly miss 🫣?

Closing PBI with any screenshots and context

**Comment by babakamyljanovssw** [comment]
> What the **most** important step to you and why?

 PBI is ready - To have clear description and AC is crucial for devs to work on PBI

> What the **least** important step to you and why? 

I couldn't choose least important step, all of them are important in the process

> **What ones do you commonly miss 🫣?**

Assigning task to myself

**Comment by jimmidier** [comment]
Getting PBI ready is the most important and closing the PBI is the least important as it can be done in a several ways but still all 3 steps are necessary to do Scrum

**Comment by JeanThirion** [comment]
> What the most important step to you and why?

Delivering - therefore done

> What the least important step to you and why?

Hard to tell... but if I had to pick I'd say getting ready, because it's the least impactful I suppose (i.e. with good comms you can fix bad readiness)

> What ones do you commonly miss 🫣?

I miss nothing.

**Comment by PennyWalker** [comment]
>What the most important step to you and why?
The Definition of done - without knowing the requirements and what they're trying to achieve it would be hard to action

      >What the least important step to you and why?
Everything is important but breaking it down is probably less important as this will happen naturally anyway - I don't think this shouldn't be done though, I think it help define the steps and make the work clearer

      >What ones do you commonly miss 🫣?
Breaking down into smaller tasks

**Comment by tino-liu** [comment]
> What the most important step to you and why?

Understanding the definition of done and also understanding acceptance criteria.

> What the least important step to you and why?

Can't think of anything that is least important.

> What ones do you commonly miss 🫣?

I don't usually create task for every little detail of work conducted on a PBI.

**Comment by ncn-ssw** [comment]
> What the most important step to you and why?
> What the least important step to you and why?

All 3 steps are equally important in my opinion:
- You need to know what you're doing
- You need to do it
- It needs to be demonstrated as done

> What ones do you commonly miss 🫣?

I occasionally miss the done comment, as I'm often playing Scrum Master and setting PBI states from the status board

**Comment by vladislav-kir** [comment]
> What the most important step to you and why?

Step 1, because that is when we confirm with the product owner that PBI is valid, so development time is not spent on something incorrect or unimportant.

> What the least important step to you and why?

All are important.

> What ones do you commonly miss 🫣?

Done Videos - I usually find that screenshots deliver just as much value, especially when PBIs are simple, which they should be...
